### PR TITLE
Add back the GUI

### PR DESCRIPTION
**Pull request by RDIL**
As discussed in the Discord, removing it was a bad idea.

- Adds back the GUI to this repo
- Edited it's csproj to work with the new build setup
- Removed duplicate rpkg generation algorithms (`generate_rpkg_quickly_from` still exists, it's just the same implementation now).  Fully backwards compatible.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -86,6 +117,7 @@ jobs:
       id: get_version
       run: echo ::set-output name=VERSION::${GITHUB_REF/refs\/tags\//}
 
+    # TODO: This action is not maintained - migrate to suitable replacement
```
Comment (AnthonyFuller) [inline .github/workflows/build.yml:120]: We can use this instead: https://github.com/softprops/action-gh-release

File: src/console.cpp [hunk 2]
```diff
@@ -40,7 +41,7 @@ int console::update_console(const std::string& message, const uint64_t indexMax,
 
 void console::display_usage_info()
 {
-    LOG("rpkg-cli v2.25.0 - Works with RPKGv1 (GKPR) and RPKGv2 (2KPR) files.");
+    LOG("rpkg-cli " << RPKG_VERSION << " - Works with RPKGv1 (GKPR) and RPKGv2 (2KPR) files.");
```
Comment (AnthonyFuller) [inline src/console.cpp:44]: Thank you for this.

File: src/generate_rpkg_from.cpp [hunk 3]
```diff
@@ -297,7 +297,13 @@ void rpkg_function::generate_rpkg_from(std::string& input_path, std::string& out
 
             uint64_t compressed_size_final;
 
-            compressed_size_final = LZ4_compress_HC(input_file_data.data(), output_file_data.data(), static_cast<int>(input_file_size), compressed_size, LZ4HC_CLEVEL_MAX);
+            if (compress)
```
Comment (AnthonyFuller) [inline src/generate_rpkg_from.cpp:300]: Surely this shouldn't compress at all if it is false?

And if that's not what the variable is supposed to name, rename it, it's very unclear.

File: src/hash_probe_from_file.cpp [hunk 4]
```diff
@@ -24,9 +23,9 @@ void rpkg_function::hash_probe_from_file(std::string& input_path, std::string& f
         LOG_AND_EXIT("Error: The folder " + input_rpkg_folder_path + " to search for RPKG files for hash probe mode does not exist.");
     }
 
-    std::string H1 = "Z:\\HITMAN1\\";
-    std::string H2 = R"(Z:\HITMAN2\Runtime\)";
-    std::string H3 = R"(C:\Program Files\Epic Games\HITMAN3\Runtime\)";
+    const std::string H1 = R"(Z:\HITMAN1\)";
```
Comment (AnthonyFuller) [inline src/hash_probe_from_file.cpp:26]: This should just be removed along side importing them from these folders.

**Review by AnthonyFuller — CHANGES_REQUESTED** [review]
General comments:
- Why does the GUI include full on DLLs such as ResourceLib and DwayneNeed _in the repo_? These surely should not be tracked.
- I didn't really review the GUI code as that's pretty much unchanged and known "working".
- Generally good, and nice to see it's gone under some clean up. I haven't tested the functions, only read through them, this build should be _extensively_ tested in nightly builds.

File: .github/workflows/build.yml (hunk 1 above)
Comment (RDIL) [reply .github/workflows/build.yml:120]: Yup.  Already had my mind on that - for a future PR.

File: src/hash_probe_from_file.cpp (hunk 4 above)
Comment (RDIL) [reply src/hash_probe_from_file.cpp:26]: cc @Notexe

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -14,25 +14,19 @@ on:
...
-      # Using 'latest' branch, the most recent CMake and ninja are installed.
-      uses: lukka/get-cmake@latest
-
-    - name: CMake cache
-      uses: actions/cache@v3
-      with:
-        path: cmake-build-release
-        key: ${{ matrix.target }}
+      run: |
+        choco install -y ninja
```
[the remark this answers is not in the document]
Comment (AnthonyFuller) [reply .github/workflows/build.yml:29]: One word: cache

**Review by AnthonyFuller — APPROVED** [review]
LGTM, thanks.

**Pull request merged**